### PR TITLE
[Merged by Bors] - feat: port LinearAlgebra.BilinearMap

### PR DESCRIPTION
**Pull request by jcommelin**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/LinearAlgebra/BilinearMap.lean [hunk 1]
```diff
@@ -0,0 +1,505 @@
...
+/-- Write out `B x y` as a sum over `B (b i) (b j)` if `b` is a basis.
+
+Version for bilinear maps, see `sum_repr_mul_repr_mulₛₗ` for the semi-bilinear version. -/
+theorem sum_repr_mul_repr_mul {B : Mₗ →ₗ[R] Nₗ →ₗ[R] Pₗ} (x y) :
+    ((b₁'.repr x).Sum fun i xi => (b₂'.repr y).Sum fun j yj => xi • yj • B (b₁' i) (b₂' j)) =
+      B x y := by
+  conv_rhs => rw [← b₁'.total_repr x, ← b₂'.total_repr y]
+  simp_rw [Finsupp.total_apply, Finsupp.sum, map_sum₂, map_sum, LinearMap.map_smul₂,
+    LinearMap.map_smul]
+#align linear_map.sum_repr_mul_repr_mul LinearMap.sum_repr_mul_repr_mul
```
Comment (eric-wieser) [inline Mathlib/LinearAlgebra/BilinearMap.lean:481]: These lemmas don't exist in mathlib3 master

**Review by eric-wieser — CHANGES_REQUESTED** [review]
Can you restart the port of this file after the next mathport run?

**Comment by eric-wieser** [comment]
I think mathport has rerun, so this is ready to re-port

**Comment by casavaca** [comment]
I re-ported this file. @jcommelin

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**